Local agent panic, better parse WOODPECKER_FILTER_LABELS option

### Component

agent

### Describe the bug

The woodpecker-agent (next) crash with the following if one does not configure correctly the `WOODPECKER_FILTER_LABELS` option:
```console
$ export WOODPECKER_FILTER_LABELS=my-local-agent
$ go run ./cmd/agent
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
main.loop(0xc0001c4380)
	/home/faust/woodpecker/cmd/agent/agent.go:64 +0x1c85
github.com/urfave/cli/v2.(*Command).Run(0xc00003e2c0, 0xc0001c4380, {0xc000190010, 0x1, 0x1})
	/home/faust/go/pkg/mod/github.com/urfave/cli/v2@v2.25.0/command.go:273 +0x9eb
github.com/urfave/cli/v2.(*App).RunContext(0xc000016000, {0x1e540d0?, 0xc000198000}, {0xc000190010, 0x1, 0x1})
	/home/faust/go/pkg/mod/github.com/urfave/cli/v2@v2.25.0/app.go:332 +0x616
github.com/urfave/cli/v2.(*App).Run(...)
	/home/faust/go/pkg/mod/github.com/urfave/cli/v2@v2.25.0/app.go:309
main.main()
	/home/faust/woodpecker/cmd/agent/main.go:42 +0x1ee
exit status 2
```

That's not a bug and it happened because of a wrong configuration, correct config is `WOODPECKER_FILTER_LABELS=agent=my-local-agent`, but IMO the agent should parse the option better and not crash since it's makes it difficult to understand where the problem is.

### System Info

```shell
docker-compose:
{"source":"https://github.com/woodpecker-ci/woodpecker","version":"next-c40a6d88"}
```


### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

--- a/cmd/agent/agent.go
+++ b/cmd/agent/agent.go
@@ -19,6 +19,7 @@ import (
 	"context"
 	"crypto/tls"
 	"errors"
+	"fmt"
 	"net/http"
 	"os"
 	"runtime"
@@ -182,9 +183,8 @@ func loop(c *cli.Context) error {
 		"repo":     "*", // allow all repos by default
 	}
 
-	for _, v := range c.StringSlice("filter") {
-		parts := strings.SplitN(v, "=", 2)
-		labels[parts[0]] = parts[1]
+	if err := stringSliceAddToMap(c.StringSlice("filter"), labels); err != nil {
+		return err
 	}
 
 	filter := rpc.Filter{
@@ -243,5 +243,23 @@ func loop(c *cli.Context) error {
 		version.String(), engine.Name(), platform, parallel)
 
 	wg.Wait()
+	return nil
+}
+
+func stringSliceAddToMap(sl []string, m map[string]string) error {
+	if m == nil {
+		m = make(map[string]string)
+	}
+	for _, v := range sl {
+		parts := strings.SplitN(v, "=", 2)
+		switch len(parts) {
+		case 2:
+			m[parts[0]] = parts[1]
+		case 1:
+			return fmt.Errorf("key '%s' does not have a value assigned", parts[0])
+		default:
+			return fmt.Errorf("empty string in slice")
+		}
+	}
 	return nil
 }

--- a/cmd/agent/agent_test.go
+++ b/cmd/agent/agent_test.go
@@ -30,11 +30,11 @@ func TestStringSliceAddToMap(t *testing.T) {
 	}{
 		{
 			name: "add values to map",
-			sl:   []string{"foo=bar", "baz=qux"},
+			sl:   []string{"foo=bar", "baz=qux=nux"},
 			m:    make(map[string]string),
 			expected: map[string]string{
 				"foo": "bar",
-				"baz": "qux",
+				"baz": "qux=nux",
 			},
 			err: false,
 		},

--- a/cmd/agent/agent_test.go
+++ b/cmd/agent/agent_test.go
@@ -1,0 +1,75 @@
+// Copyright 2023 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStringSliceAddToMap(t *testing.T) {
+	tests := []struct {
+		name     string
+		sl       []string
+		m        map[string]string
+		expected map[string]string
+		err      bool
+	}{
+		{
+			name: "add values to map",
+			sl:   []string{"foo=bar", "baz=qux"},
+			m:    make(map[string]string),
+			expected: map[string]string{
+				"foo": "bar",
+				"baz": "qux",
+			},
+			err: false,
+		},
+		{
+			name:     "empty slice",
+			sl:       []string{},
+			m:        make(map[string]string),
+			expected: map[string]string{},
+			err:      false,
+		},
+		{
+			name:     "missing value",
+			sl:       []string{"foo", "baz=qux"},
+			m:        make(map[string]string),
+			expected: map[string]string{},
+			err:      true,
+		},
+		{
+			name:     "empty string in slice",
+			sl:       []string{"foo=bar", "", "baz=qux"},
+			m:        make(map[string]string),
+			expected: map[string]string{},
+			err:      true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := stringSliceAddToMap(tt.sl, tt.m)
+
+			if tt.err {
+				assert.Error(t, err)
+			} else {
+				assert.EqualValues(t, tt.expected, tt.m)
+			}
+		})
+	}
+}

